Bottom: 933efac75b2970b378a9362565c1c9539ee70d35
Top:    4b6adce26ba8f646a5c258134794671aa8803aba
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-09-30 22:45:36 +0100

Ensure "Gathering" mode is set after painting the screen.

In between repaints initiated from the GUI, with this flag set, all
Draw{object_type} operations just update a dirtied bounds. A final
Draw() call then calls the HID to invalidate the bounds gathered.

Setting "Gathering = True" after an expose event callback ensures
subsequent drawing is batched up for the next screen update.


---

diff --git a/src/draw.c b/src/draw.c
index 96ab0b8..50e6fef 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -2596,4 +2596,6 @@ hid_expose_callback (HID * hid, BoxType * region, void *item)
   Output.fgGC = savefg;
   Output.bgGC = savebg;
   Output.pmGC = savepm;
+
+  Gathering = True;
 }
